dask/bag/core.py

```diff
@@ -183,6 +183,42 @@ def finalize_item(results):
     return results[0]
 
 
+def merge_kwargs(dsk, kwargs):
+    """ Merges dask values from kwargs with dsk.
+
+    Returns the merged dsk and a graph expression useful for use with apply in
+    a dask graph to pass the final computed **kwargs to a function.
+
+    Currently only dask.bag.Item and python literal values are supported.
+    """
+    kw_pairs = []
+    for key, val in iteritems(kwargs):
+        if isinstance(val, Item):
+            dsk = merge(dsk, val.dask)
+            val = val.key
+        # TODO elif isinstance(val, Value):
+        elif isinstance(val, Base):
+            raise NotImplementedError(
+                '%s not supported as kwarg value to Bag.map_partitions'
+                % type(val).__name__)
+        kw_pairs.append([key, val])
+    return dsk, kw_pairs
```

Comment:
I'm not certain that we actually want the full machinery of `to_task_dasks` here, unless we really do want people to be able to pass fully-realized bags, dataframes, arrays, etc along... which seems like a good way to get horrible performance presuming you don't blow out memory ;-)

But let's say we do:
-`to_task_dasks` is from the imperative side at present, it only treats `Value`s natively (everything else is optimize/inlined) which causes expensive computations (e.g. shard loading) to get duplicated in my realistic two-pass use-case
- this was why in https://github.com/dask/dask/pull/1041/commits I first factored a generalized core out of the imperative version, and then introduced a bag-native version reusing the same python-object-walking core

Maybe what's needed is more clarity around the `to_task_dasks` work started in #1041.  For example rather than the weird, but Works For Me "special" support of the [new `base.normalize_to_dasks` core](https://github.com/dask/dask/pull/1041/commits/9a818bfdcd306659c29dd18ac6bb93afb008c715):
  - it should have a way to differentiate between which dask types get referenced by key (error if a referenceable type has more than one key) vs optimized and inlined
  - it could have a way to blacklist certain types (presuming there's no better way to say "all large dasks" if that's a thing we want)